src/coreclr/vm/runtimehandles.cpp

```diff
@@ -2622,6 +2622,19 @@ FCIMPL1(ReflectClassBaseObject*, RuntimeFieldHandle::GetApproxDeclaringType, Fie
 }
 FCIMPLEND
 
+FCIMPL1(void*, RuntimeFieldHandle::GetApproxFieldType, FieldDesc *pField) {
+    CONTRACTL {
+        FCALL_CHECK;
+    }
+    CONTRACTL_END;
+
+    if (!pField)
+        FCThrowRes(kArgumentNullException, W("Arg_InvalidHandle"));
+
+    return pField->GetApproxFieldTypeHandleThrowing().GetMethodTable();
```

Comment:
I'm particularly unsure with this FCall:

- Get field type is marked as throwing. When will it throw? If the type fails to load?
- Will the type of RVA field ever be a `TypeDesc`?
- What's the overhead of creating `RuntimeType` from `TypeHandle`?
- There is managed equivalent for this (`RtFieldInfo.InitializeFieldType`), with managed `Signature`. What about its overhead?

Reply:
> Get field type is marked as throwing. When will it throw? If the type fails to load?

Yes.

> Will the type of RVA field ever be a TypeDesc?

Yes.

> What's the overhead of creating RuntimeType from TypeHandle?

One time object allocation that will stay around till end of the process (or till the collectible ALC is unloaded).

> There is managed equivalent for this (RtFieldInfo.InitializeFieldType), with managed Signature. What about its overhead?

RtFieldInfos and RuntimeMethodInfo are heavy reflection. Creating them has a lot of overhead.

Reply:
> RtFieldInfos and RuntimeMethodInfo are heavy reflection. Creating them has a lot of overhead.

I considered to borrow the Signature construction, not creating the full RtFieldInfo type. Extra object allocations can be a hit.

Though RuntimeType allocation is one time, it will likely never be used in other places. One object per array literal site may also count for the whole program.

Maybe just returning `LoadSize` is most performant here, but I'm worrying about reusability of a new FCall.

Reply:
> I'm worrying about reusability of a new FCall.

Minimizing number of FCalls/QCalls is secondary concern. We care about:

- Good performance
- Reasonable FCall/QCall boundary. Avoid trickly contracts.
- Avoiding HELPER_METHOD_FRAME and FCThrow in FCall implementation. We want to get rid of them. (#95695)